Benchmark/hop_count/kilobot.cpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include "../../../Swarm-robotics-benchmark/Driver/Kilobot_simulation/kilobot_driver.h"
+
+START_USER_PROGRAM
+
+Channel *channel;
+Publisher *publisher;
+Subscriber *subscriber; 
+unsigned char msg1[10];
+
+int minhop1 = -1;
+int minhop2 = -1;
+
+void sent() {
+    publisher->send(msg1, strlen((char*)msg1));
+}
+
+void call_back(unsigned char * msg, int size, int hop, Meta_t * meta) {
+    if (msg[0] == 'A'){
+        if (hop > minhop1){
+            minhop1 = hop;
+        }  
+    }
+    if (msg[0] == 'B'){
+        if (hop > minhop2){
+            minhop2 = hop;
+        }  
+    }
+} 
+
+void loop() {
+    if (id == 3) {

```

Comment:
Instead of using id to differentiate, use separate user files for each root and followers